src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -0,0 +1,574 @@
+        systemMessageObject.Insert(L"role", WDJ::JsonValue::CreateStringValue(L"system"));
+        systemMessageObject.Insert(L"content", WDJ::JsonValue::CreateStringValue(systemMessageContent));
+        _jsonMessages.Append(systemMessageObject);
+        _queryBox().Focus(FocusState::Programmatic);
+    }
+
+    // Method Description:
+    // - This event is called when the user clicks on a Chat Message. We will
+    //   dispatch the contents of the message to the app to input into the active control.
+    // Arguments:
+    // - e: an ItemClickEventArgs who's ClickedItem() will be the message that was clicked on.
+    // Return Value:
+    // - <none>
+    void ExtensionPalette::_listItemClicked(const Windows::Foundation::IInspectable& /*sender*/,
+                                            const Windows::UI::Xaml::Controls::ItemClickEventArgs& e)
+    {
+        const auto selectedSuggestionItem = e.ClickedItem();
+        const auto selectedItemAsChatMessage = selectedSuggestionItem.as<winrt::Microsoft::Terminal::Query::Extension::ChatMessage>();
+        if (selectedItemAsChatMessage.IsCode())
+        {
+            auto suggestion = winrt::to_string(selectedItemAsChatMessage.MessageContent());
+
+            // the AI sometimes sends code blocks with newlines in them
+            // sendInput doesn't work with single new lines, so we replace them with \r
+            size_t pos = 0;
+            while ((pos = suggestion.find("\n", pos)) != std::string::npos)
```

Comment:
yikes what happens if it hands back `\r\n`'s though, does that work? (or does this generate `\r\r`)

Reply:
Yep it will turn it into `\r\r`. I have yet to see a case where it returns `\r` though, it only ever returns `\n`.